heat/core/tensor.py

```diff
@@ -1513,24 +1569,254 @@ def __repr__(self, *args):
         return self.__array.__repr__(*args)
 
     def __getitem__(self, key):
-        # TODO: document me
-        # TODO: test me
-        # TODO: sanitize input
-        # TODO: make me more numpy API complete
-        return tensor(self.__array[key], self.shape, self.split, self.device, self.comm)
+        """
+        Global getter function for ht.tensors
+
+        Parametes
+        ---------
+        key : int, slice, tuple, list
+            indices to get from the tensor.
+
+        Returns
+        -------
+        result : ht.tensor
+            getter returns a new ht.tensor composed of the elements of the original tensor selected by the indices given.
+            this does *NOT* redistribute or rebalance the resulting tensor. If the selection of values is unbalanced then
```

Comment:
Could you please add this to the balance() issue as a note, that getters and setters need to be rebalanced OR open a separate issue referencing balance()